File: 3154-maximum-value-of-an-ordered-triplet-i/maximum-value-of-an-ordered-triplet-i.cs
public class Solution {
    public long MaximumTripletValue(int[] nums) {
        int ml = nums[0], md=0;
        long ans=0;
        for(int i=1; i<nums.Length; i++){
            ans=Math.Max(ans, (long)md*nums[i]);
            md=Math.Max(md, ml-nums[i]);
            ml=Math.Max(ml, nums[i]);
        }
        return ans;
    }
}

File: 3154-maximum-value-of-an-ordered-triplet-i/maximum-value-of-an-ordered-triplet-i.java
class Solution {
    public long maximumTripletValue(int[] nums) {
        int ml=nums[0], md=0;
        long ans=0;
        for(int i=1; i<nums.length; i++){
            ans=Math.max(ans, (long)md*nums[i]);
            md=Math.max(md, ml-nums[i]);
            ml=Math.max(ml, nums[i]);
        }
        return ans;
    }
}